src/main/java/com/feedzai/commons/sql/abstraction/batch/AbstractBatch.java

```diff
@@ -399,7 +436,9 @@ public void flush() {
             logger.trace("[{}] Batch flushed. Took {} ms, {} rows.", name, (System.currentTimeMillis() - start), temp.size());
         } catch (final Exception e) {
             if (this.maxFlushRetries > 0) {
-                logger.warn(dev, "[{}] Error occurred while flushing. Retrying.", name, e);
+                final String msg = "[{}] Error occurred while flushing. Retrying.";
+                logger.warn(dev, msg, name);
+                confidentialLogger.warn(dev, msg, name, e);
```

Comment:
I added the checks. I was going to create a method to centralize that logic, but several logger methods are called with different types of parameters. So I decided to add a simple if block to prevent duplicate logging when a confidential logger is not provided.